addons/account/models/account_bank_statement.py

```diff
@@ -735,18 +758,24 @@ def _prepare_counterpart_move_line_vals(self, counterpart_vals, move_line=None):
 
                 if self.foreign_currency_id == company_currency:
                     amount_currency = 0.0
-                else:
+                elif statement_line_rate:
                     amount_currency = company_currency._convert(balance, journal_currency, journal.company_id, self.date)
                     amount_currency *= statement_line_rate
                     currency_id = self.foreign_currency_id.id
+                else:
+                    amount_currency = amount_residual_currency
+                    currency_id = self.foreign_currency_id.id
 
             elif self.foreign_currency_id and not journal_currency:
 
                 # Different currencies set on the company and the statement line.
                 # Record the counterpart line using the foreign currency.
 
-                amount_currency = balance * statement_line_rate
                 currency_id = self.foreign_currency_id.id
+                if statement_line_rate:
+                    amount_currency = balance * statement_line_rate
+                else:
+                    amount_currency = amount_residual_currency
```

Comment:
All those else seem to always assign amount_residual_currency, right ? Wouldn't there be a way to make it a single condition instead of repeating the block every time ?

Also, some comment explaining why this condition on statement_line_rate is necessary would probably be welcome